src/main/java/com/conveyal/gtfs/stats/StopStats.java

```diff
@@ -1,7 +1,239 @@
+                continue;
+            }
+
+            timesAtStop.add(st.departure_time);
+        }
+        timesAtStop.sort();
+
+        // convert to deltas
+        TIntList deltas = new TIntArrayList();
+
+        for (int i = 0; i < timesAtStop.size() - 1; i++) {
+            int delta = timesAtStop.get(i + 1) - timesAtStop.get(i);
+
+            if (delta > 60) deltas.add(delta);
+        }
+
+        if (deltas.isEmpty()) return -1;
+
+        return deltas.sum() / deltas.size();
+    }
+
+    public int getHeadwayForStopByRoute (String stop_id, String route_id, LocalDate date, LocalTime from, LocalTime to) {
+
+        List<Trip> tripsForStop = feed.getStopTimesForStop(stop_id).stream()
+                .filter(t -> feed.trips.get(t.a).route_id.equals(route_id))
+                .map(t -> feed.trips.get(t.b.a))
+                .filter(trip -> feed.services.get(trip.service_id).activeOn(date))
+                .collect(Collectors.toList());
+
+        return getStopHeadwayForTrips(stop_id, tripsForStop, from, to);
+    }
+
+    /** Returns a map from route_id to transferPerformanceSummary for each route at a stop for a specified date. */
+    public List<TransferPerformanceSummary> getTransferPerformance (String stop_id, LocalDate date) {
+        SortedSet<Fun.Tuple2<String, Fun.Tuple2>> stopTimes = feed.getStopTimesForStop(stop_id);
+        Map<String, List<StopTime>> routeStopTimeMap  = new HashMap<>();
+        List<TransferPerformanceSummary> transferPerformanceMap = new ArrayList<>();
+        // TODO: do we need to handle interpolated stop times???
+
+        // first stream stopTimes for stop into route -> list of stopTimes map
+        stopTimes.stream()
+                .forEach(t -> {
+                    Trip trip = feed.trips.get(t.b.a);
+                    Service service = feed.services.get(trip.service_id);
+                    // only add to map if trip is active on given date
+                    if (service != null && service.activeOn(date)) {
+                        StopTime st = feed.stop_times.get(t.b);
+                        Route route = feed.routes.get(trip.route_id);
+
+                        List<StopTime> times = new ArrayList<>();
+                        if (routeStopTimeMap.containsKey(route.route_id)) {
+                            times.addAll(routeStopTimeMap.get(route.route_id));
+                        }
+                        times.add(st);
+                        routeStopTimeMap.put(route.route_id, times);
+                    }
+                });
+        Map<Fun.Tuple2<String, String>, TIntList> waitTimesByRoute = new HashMap<>();
+        Map<Fun.Tuple2<String, String>, Set<Fun.Tuple2<StopTime, StopTime>>> missedTransfers = new HashMap<>();
+        // iterate over every entry in route -> list of stopTimes map
+        for (Map.Entry<String, List<StopTime>> entry : routeStopTimeMap.entrySet()) {
+            final int MISSED_TRANSFER_THRESHOLD = 60 * 10;
```

Comment:
Yeah probably, although I wonder if there's a way to specify this without using an arbitrary parameter like this.